lang/test/org/partiql/lang/planner/memorydb/QueryEngine.kt

```diff
@@ -0,0 +1,211 @@
+package org.partiql.lang.planner.memorydb
+
+import com.amazon.ionelement.api.toIonValue
+import org.partiql.lang.ION
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.eval.BindingCase
+import org.partiql.lang.eval.BindingName
+import org.partiql.lang.eval.Bindings
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.StructOrdering
+import org.partiql.lang.eval.namedValue
+import org.partiql.lang.planner.DmlAction
+import org.partiql.lang.planner.GlobalResolutionResult
+import org.partiql.lang.planner.GlobalVariableResolver
+import org.partiql.lang.planner.PlannerPassResult
+import org.partiql.lang.planner.PlannerPipeline
+import org.partiql.lang.planner.QueryResult
+import org.partiql.lang.planner.StaticTypeResolver
+import org.partiql.lang.planner.memorydb.operators.GetByKeyProjectRelationalOperatorFactory
+import org.partiql.lang.planner.transforms.optimizations.createFilterScanToKeyLookupPass
+import org.partiql.lang.planner.transforms.optimizations.createRemoveUselessAndsPass
+import org.partiql.lang.planner.transforms.optimizations.createRemoveUselessFiltersPass
+import org.partiql.lang.types.BagType
+import org.partiql.lang.types.StructType
+import org.partiql.lang.util.SexpAstPrettyPrinter
+import org.partiql.pig.runtime.DomainNode
+import java.util.UUID
+
+// The name of the database in the context variable.
+internal const val DB_CONTEXT_VAR = "in-memory-database"
+
+/**
+ * This class is a demonstration of how to integrate a storage layer with PartiQL's query planner.
+ */
+class QueryEngine(val db: MemoryDatabase) {
+    var enableDebugOutput = false
```

Comment:
nit: could include `enableDebugOutput` as an argument to the function and make it immutable since it's not changed in the existing code

Reply:
I would prefer to leave this public and mutable since it spits out hundreds of lines per query its mutability allows me to enable it only for one query out of a sequence.